sycl/include/sycl/sycl.hpp

```diff
@@ -8,6 +8,30 @@
 
 #pragma once
 
+// Throw warning when including sycl.hpp without using -fsycl flag.
+// Warning can be disabled by defining SYCL_DISABLE_FSYCL_SYCLHPP_WARNING macro.
+#define STRINGIFY(x) #x
+#define TOSTRING(x) STRINGIFY(x)
+
+#ifdef _MSC_VER
```

Comment:
Note to reviewers: Here's a working example of warnings being thrown for different compilers: https://godbolt.org/z/bvoP3q3Gx